tests/src/JIT/Performance/CodeQuality/BenchI/NDhrystone/NDhrystone.cs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+// Adapted from
+//
+// Dhrystone: a synthetic systems programming benchmark
+// Reinhold P. Weicker
+// Communications of the ACM, Volume 27 Issue 10, Oct 1984, Pages 1013-1030
+ 
+using Microsoft.Xunit.Performance;
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+[assembly: OptimizeForBenchmarks]
+[assembly: MeasureInstructionsRetired]
+
+public static class NDhrystone
+{
+
+#if DEBUG
+    public const int Iterations = 1;
+#else
+    public const int Iterations = 7000000;
+#endif
+
+    static T[][] AllocArray<T>(int n1, int n2) {
+        T[][] a = new T[n1][];
+        for (int i = 0; i < n1; ++i) {
+            a[i] = new T[n2];
+        }
+        return a;
+    }
+
+    enum Enumeration
+    {
+        Ident1 = 1, Ident2, Ident3, Ident4, Ident5
+    }
+
+    sealed class Record
+    {
+        public Record PtrComp;
+        public Enumeration Discr;
+        public Enumeration EnumComp;
+        public int IntComp;
+        public char[] StringComp;
+    }
+
+    static int s_intGlob;
+    static bool s_boolGlob;
+    static char s_char1Glob;
+    static char s_char2Glob;
+    static int[] m_array1Glob = new int[51];
+    static int[][] m_array2Glob;
+    static Record m_ptrGlb = new Record();
+    static Record m_ptrGlbNext = new Record();
+    static char[] m_string1Loc;
+    static char[] m_string2Loc;
+
+    static void Proc0() {
+        int intLoc1;
+        int intLoc2;
+        int intLoc3 = 0;
+        Enumeration enumLoc;
+        
+        int i;   /* modification */
+        
+        m_ptrGlb.PtrComp = m_ptrGlbNext;
+        m_ptrGlb.Discr = Enumeration.Ident1;
+        m_ptrGlb.EnumComp = Enumeration.Ident3;
+        m_ptrGlb.IntComp = 40;
+        m_ptrGlb.StringComp = "DHRYSTONE PROGRAM, SOME STRING".ToCharArray();
+        m_string1Loc = "DHRYSTONE PROGRAM, 1'ST STRING".ToCharArray();
+        m_array2Glob[8][7] = 10;  /* Was missing in published program */
+
+        for (i = 0; i < Iterations; ++i) {
+            Proc5();
+            Proc4();
+            intLoc1 = 2;
+            intLoc2 = 3;
+            m_string2Loc = "DHRYSTONE PROGRAM, 2'ND STRING".ToCharArray();
```

Comment:
Curious why you rewrote this to note use inline assignments? Not against it, but might have been more useful to test the optimizer the way it was before (successive array stores).